ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -727,6 +741,31 @@ void setChunkException(RouterException exception) {
     }
 
     /**
+     * Slice this chunk's data to only include the bytes within the operation's specified byte range.
+     * @param blobData the {@link BlobData} for this chunk.
+     * @return A {@link ByteBuffer} that only includes bytes within the operation's specified byte range.
+     */
+    ByteBuffer filterChunkToRange(BlobData blobData) {
+      ByteBuffer buf = blobData.getStream().getByteBuffer();
+      if (!hasRange) {
+        return buf;
+      }
+      GetBlobRange range = options.getRange();
+      long startOffset = range.getStartOffset(totalSize);
+      long endOffset = range.getEndOffset(totalSize);
```

Comment:
Also why do this every time? `startOffset` and `endOffset` could be set just once when the `firstChunk` is deserialized (in the overridden `handleBody()` method).